File: Python/Distribute Candies.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(n)
    """

    def distributeCandies(self, candy_types: List[int]) -> int:
        unique_candies = len(set(candy_types))
        return min(unique_candies, len(candy_types) // 2)

File: Java/Distribute Candies.java
import java.util.HashSet;
import java.util.Set;

public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(n)
     */
    public int distributeCandies(int[] candyType) {
        Set<Integer> uniqueCandiesSet = new HashSet<>();

        for (int candy : candyType)
            uniqueCandiesSet.add(candy);

        return Math.min(uniqueCandiesSet.size(), candyType.length / 2);
    }
}
